internal/plugin.go

```diff
@@ -1209,121 +1101,153 @@ func (p *Plugin) assertValidConfig() error {
+				"%d PlacementRule",
+			plCount.plc, plCount.plr,
+		)
+	}
 
-		if len(plcset.Placement.ClusterSelectors) > 0 && len(plcset.Placement.LabelSelector) > 0 {
-			return fmt.Errorf(
-				"policySet %s must provide only one of placement.labelSelector or placement.clusterselectors",
-				plcset.Name,
-			)
-		}
+	p.usingPlR = plCount.plc == 0
 
-		// validate placement names are DNS compliant
-		plcSetPlrName := plcset.Placement.PlacementRuleName
-		if plcSetPlrName != "" && len(validation.IsDNS1123Subdomain(plcSetPlrName)) > 0 {
-			return fmt.Errorf(
-				"plcset.Placement.PlacementRuleName `%s` is not DNS compliant. See %s", plcSetPlrName, dnsReference,
-			)
-		}
+	return nil
+}
 
-		plcSetPlcmtPlName := plcset.Placement.PlacementName
-		if plcSetPlcmtPlName != "" && len(validation.IsDNS1123Subdomain(plcSetPlcmtPlName)) > 0 {
-			return fmt.Errorf(
-				"plcset.Placement.PlacementName `%s` is not DNS compliant. See %s", plcSetPlcmtPlName, dnsReference,
-			)
-		}
+// assertValidPlacement is a helper for assertValidConfig to verify placement configurations
+func assertValidPlacement(
+	placement types.PlacementConfig,
+	generatePlacement bool,
+	path string,
+	plCount *struct {
+		plc int
+		plr int
+	},
+) error {
+	if placement.PlacementRulePath != "" && placement.PlacementPath != "" {
+		return fmt.Errorf(
+			"%s must provide only one of placement.placementPath or placement.placementRulePath", path,
+		)
+	}
 
-		plcSetPlName := plcset.Placement.Name
-		if plcSetPlName != "" && len(validation.IsDNS1123Subdomain(plcSetPlName)) > 0 {
-			return fmt.Errorf(
-				"plcset.Placement.Name `%s` is not DNS compliant. See %s", plcSetPlName, dnsReference,
-			)
-		}
+	if len(placement.ClusterSelectors) > 0 && len(placement.LabelSelector) > 0 {
+		return fmt.Errorf(
+			"%s must provide only one of placement.labelSelector or placement.clusterSelectors", path,
+		)
+	}
 
-		policySetPlacementOptions := 0
-		if len(plcset.Placement.LabelSelector) != 0 || len(plcset.Placement.ClusterSelectors) != 0 {
-			policySetPlacementOptions++
-		}
+	if placement.PlacementRuleName != "" && placement.PlacementName != "" {
+		return fmt.Errorf(
+			"%s must provide only one of placement.placementName or placement.placementRuleName", path,
+		)
+	}
 
-		if plcset.Placement.PlacementRulePath != "" || plcset.Placement.PlacementPath != "" {
-			policySetPlacementOptions++
-		}
+	defaultPlacementOptions := 0
+	if len(placement.LabelSelector) != 0 || len(placement.ClusterSelectors) != 0 {
+		defaultPlacementOptions++
+	}
 
-		if plcset.Placement.PlacementRuleName != "" || plcset.Placement.PlacementName != "" {
-			policySetPlacementOptions++
-		}
+	if placement.PlacementRulePath != "" || placement.PlacementPath != "" {
+		defaultPlacementOptions++
+	}
+
+	if placement.PlacementRuleName != "" || placement.PlacementName != "" {
+		defaultPlacementOptions++
+	}
+
+	if defaultPlacementOptions > 0 && !generatePlacement {
```

Comment:
I don't necessarily think we should change this, but I was a bit surprised that I wouldn't be able to override a default placement from `policyDefaults` being applied to my policy/policy set by setting `generatePlacement: false` explicitly on it.

Reply:
True--I suppose even if a placement is specified it becomes irrelevant when the `generatePlacement: false` flag is set. Especially in GitOps, it probably makes more sense here to skip this check and let things deploy. I'll take it out.

Reply:
Okay, now a placement can be specified but setting `generatePlacement` explicitly will take priority and not throw an error.